src/utils.jl

```diff
@@ -288,3 +261,58 @@ function range_shuffle!(r::UnitRange, a::AbstractVector; seed::Int=-1)
     end
 end
 
+"""
+    randbn(n, p, seed=-1)
+
+Return a binomally-distribted random number with parameters `n` and `p` and optional `seed`.
+
+### References
+- "Non-Uniform Random Variate Generation," Luc Devroye, p. 522. Retrieved via http://www.eirene.de/Devroye.pdf.
+- http://stackoverflow.com/questions/23561551/a-efficient-binomial-random-number-generator-code-in-java
+"""
+function randbn(n::Integer, p::Real, rng::AbstractRNG)
+    log_q = log(1.0 - p)
+    x = 0
+    sum = 0.0
+    while true
+        sum += log(rand(rng)) / (n - x)
+        sum < log_q && break
+        x += 1
+    end
+    return x
+end
+
+"""
+    is_graphical(degs)
+
+Return `true` if the degree sequence `degs` is graphical.
+A sequence of integers is called graphical, if there exists a graph where the degrees of its vertices form that same sequence.
+
+### Performance
+Time complexity: ``\\mathcal{O}(|degs|*\\log(|degs|))``.
+
+### Implementation Notes
+According to Erdös-Gallai theorem, a degree sequence ``\\{d_1, ...,d_n\\}`` (sorted in descending order) is graphical iff the sum of vertex degrees is even and the sequence obeys the property -
+```math
+\\sum_{i=1}^{r} d_i \\leq r(r-1) + \\sum_{i=r+1}^n min(r,d_i)
+```
+for each integer r <= n-1
+"""
+function is_graphical(degs::Vector{<:Integer})
+    iseven(sum(degs)) || return false
+    sorted_degs = sort(degs, rev = true)
+    n = length(sorted_degs)
+    cur_sum = zero(UInt64)
+    mindeg = Vector{UInt64}(undef, n)
+    @inbounds for i = 1:n
+        mindeg[i] = min(i, sorted_degs[i])
+    end
+    cum_min = sum(mindeg)
+    @inbounds for r = 1:(n - 1)
+        cur_sum += sorted_degs[r]
+        cum_min -= mindeg[r]
+        cond = cur_sum <= (r * (r - 1) + cum_min)
+        cond || return false
+    end
+    return true
+end
```

Comment:
That's it for the code! Everything below here is test files, so feel free to stop here or you can look to see how we've modified the tests to use the new structures.